### PR TITLE
quirc QR code decoding library

### PR DESCRIPTION
**Pull request by igrr**
# Checklist

- [x] Component contains License
- [x] Component contains README.md
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [x] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [x] _Optional:_ Component contains unit tests
- [x] CI passing

# Change description

This PR adds a QR code decoding library, [quirc](https://github.com/dlbeer/quirc). Aside from the large stack size required, it can be used in ESP-IDF without modifications.

### CODE REVIEW

File: quirc/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Quirc QR code decoding library
+
```
Comment (tore-espressif) [inline quirc/README.md:2]: Let's enhance the readme with version badge:

```suggestion

[![Component Registry](https://components.espressif.com/components/espressif/quirc/badge.svg)](https://components.espressif.com/components/espressif/quirc)

```

File: quirc/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+This is an ESP-IDF component for [quirc](https://github.com/dlbeer/quirc), a QR code decoding library.
+
```
Comment (tore-espressif) [inline quirc/README.md:4]: Would it make sense to cross-reference `qrcode` component for QR code generation?

**Review by tore-espressif — APPROVED** [review]
@igrr Only minor comments in README, otherwise LGTM!

**Comment by dmitry1945** [comment]
@igrr LGTM!

**Comment by igrr** [comment]
@tore-espressif Thanks for the review, have addressed your comments.
@dmitry1945 Thanks for taking a look as well.

**Pull request merged**